src/hotspot/cpu/aarch64/codeBuffer_aarch64.cpp

```diff
@@ -26,6 +26,55 @@
 #include "asm/codeBuffer.inline.hpp"
 #include "asm/macroAssembler.hpp"
 
+void CodeBuffer::shared_stub_to_runtime_for(address dest, int caller_offset) {
+  if (_shared_stub_to_runtime_call_requests == nullptr) {
+    _shared_stub_to_runtime_call_requests = new SharedStubToRuntimeCallRequests();
+  }
+  SharedStubToRuntimeCallRequest request(dest, caller_offset);
+  _shared_stub_to_runtime_call_requests->push(request);
+  _finalize_stubs = true;
+}
+
+template <typename MacroAssembler>
```

Comment:
Why does this need to be a template? It seems to only ever be instantiated with one concrete class (the real MacroAssembler).